src/hotspot/share/asm/codeBuffer.hpp

```diff
@@ -273,70 +272,75 @@ class CodeSection {
 #endif //PRODUCT
 };
 
-class CodeString;
-class CodeStrings {
-private:
+
 #ifndef PRODUCT
-  CodeString* _strings;
-  CodeString* _strings_last;
-#ifdef ASSERT
-  // Becomes true after copy-out, forbids further use.
-  bool _defunct; // Zero bit pattern is "valid", see memset call in decode_env::decode_env
-#endif
-  static const char* _prefix; // defaults to " ;; "
 
-  CodeString* find(intptr_t offset) const;
-  CodeString* find_last(intptr_t offset) const;
+class AsmRemarkCollection;
+class DbgStringCollection;
 
-  void set_null_and_invalidate() {
-    _strings = NULL;
-    _strings_last = NULL;
-#ifdef ASSERT
-    _defunct = true;
-#endif
-  }
-#endif
+// The assumption made  here is that most code remarks  (or comments) added to
+// the generated assembly  code are unique, i.e. there is  very little gain in
+// trying to  share the  strings between  the different  offsets tracked  in a
+// buffer (or blob).
```

Comment:
Noticed some double whitespaces in many of the code comments, for example "assembly__code", "different__offsets".

Reply:
Comments are (fully) justified for readability (within limits, not using hyphenation), same as text in books, newspapers and reports.

Reply:
As we discussed off-thread, I would prefer to not manually enforce block text layout by inserting extra whitespace because it is tedious and increases the burden on anyone later modifying (parts of) that comment.

Although I do understand that this layout might be easier to read for some people, I think it's the IDE's responsibility to display the comment in the user preferred way. If at all, line length restrictions and such should be part of the HotSpot Style Guide.